### PR TITLE
fix(integrated_channels): :bug: fix forward for parameter rename bug

### PR DESCRIPTION
**Pull request by edx-abolger**
Caused in 3.26.7

**Merge checklist:**
- [n/a ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [n/a ] `make static` has been run to update webpack bundling if any static content was updated
- [n/a] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [n/a ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py [hunk 1]
```diff
@@ -38,6 +39,23 @@ def setUp(self):
...
+            'username': "fake_username",
+            'course_run_id': "TEST_COURSE_RUN_KEY"
+        }
+        # If you change the arg names to transmit_single_learner_data
+        # this test will now fail with a TypeError:
+        try:
+            transmit_single_learner_data(**edx_platform_api_signal_kwargs)
+        except TypeError:
+            assert False  # if we see a Type error, the test should fail because we failed with bad keyword parameters
+        except Exception:   # pylint: disable=broad-except
```
Comment (taliaedX) [inline tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py:56]: This feels like a weird pass condition. I'm also generally not a fan of highly specific tests that only really avoid 1 scenario we've already hit. Ideally I'd rather us actually test the logic of this method but yeah that's probably nightmarish.

**Review by taliaedX — APPROVED** [review]
I'm really not a fan of the test, but I don't have better suggestions without doing some substantial digging and as we want this to stop failing asap, I'll approve it.

File: tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py (hunk 1 above)
Comment (edx-abolger) [reply tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py:56]: Oh this does not test the function at all: the problem is that unlike for REST API violations, no test exists to catch if we violate the plugin interface with the wrong params.

Comment (iloveagent57) [reply tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py:56]: You could do something like the converse: call it with the wrong keyword(s) and do a `with self.assertRaises(...):`

Comment (taliaedX) [reply tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py:56]: That wouldn't have caught this issue though would it? The issue was that we were in fact passing the wrong keywords. Thus a test that asserts that error gets raised is saying "yes this fails because these keywords are wrong and we don't want them. That is correct and thus this test passes!" would also have passed before this code change?

**Comment by macdiesel** [comment]
FWIW I don't think a test is necessary here and I would advocate for not adding one at all to cover this edge case. Changing APIs (in this case the method signature) is a risky proposition. Adding a test to cover this doesn't really improve the testing and makes the runtime longer. Both things we should be avoiding. 

For me the take away on this is that changing method signatures is risky and not high value. I think that adding a more detailed doc string would have solved the same problem. We should  avoid doing so in the future unless we have a really compelling case to do so and if we make that decision we should grep the other code bases that depend on this function/method/API. 

Last thought, when API changes like this are made, it should be a major version update (the number to the left of the dot) and not a minor version upgrade as suggested by [semantic versioning.](https://semver.org/)

**Comment by macdiesel** [comment]
For edX Semantic Versioning is covered under [OEP-47](https://open-edx-proposals.readthedocs.io/en/latest/oep-0047-bp-semantic-versioning.html)

File: tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py (hunk 1 above)
Comment (edx-abolger) [reply tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_learner_data.py:56]: @taliaedX you are correct: the goal of the test was defensive coding that would raise an error if someone did what I did and changed a method signature. However, I do like @macdiesel 's solution going forward of updating the doc strings to note that the method signatures are actually an API contract instead, as that provides defensive coding right where the programmer can see it. I will make another small doc-only PR with a docstring note in anything in integrated channels we call as a kwarg.

**Comment by edx-abolger** [comment]
> FWIW I don't think a test is necessary here and I would advocate for not adding one at all to cover this edge case. Changing APIs (in this case the method signature) is a risky proposition. Adding a test to cover this doesn't really improve the testing and makes the runtime longer. Both things we should be avoiding.
> 
> For me the take away on this is that changing method signatures is risky and not high value. I think that adding a more detailed doc string would have solved the same problem. We should avoid doing so in the future unless we have a really compelling case to do so and if we make that decision we should grep the other code bases that depend on this function/method/API.
> 
> Last thought, when API changes like this are made, it should be a major version update (the number to the left of the dot) and not a minor version upgrade as suggested by [semantic versioning.](https://semver.org/)

Would the major bump you're advocating for here have been when I changed the signature in the first place? In retrospect I agree, (had I realized it was an interface change) but the bug fix to put it back I still consider a patch? I suppose semantic versioning can't really have opinions beyond 'always bump the version' about  "versions of code you intend to not have any breaking changes but actually do ...".

**Comment by macdiesel** [comment]
> Would the major bump you're advocating for here have been when I changed the signature in the first place? 

Correct.

>In retrospect I agree, (had I realized it was an interface change) but the bug fix to put it back I still consider a patch? I suppose semantic versioning can't really have opinions beyond 'always bump the version' about "versions of code you intend to not have any breaking changes but actually do ...".

Yeah it kind of ends up in strange territory. I think that the minor version bump for the fix is fine since the API change was pulled back.
